fog/test_infra/src/db_tests.rs

```diff
@@ -169,66 +166,24 @@ pub fn recovery_db_smoke_tests_new_apis<DB: RecoveryDb>(
 }
 
 // Basic tests that missed blocks reporting works as expected
-pub fn recovery_db_missed_blocks_reporting(_rng: &mut impl RngCore, db: &impl RecoveryDb) {
-    let (user_events, _next_start_from_user_event_id) = db.search_user_events(0).unwrap();
-    assert_eq!(user_events.len(), 0);
-
-    let missed_block_ranges = db.get_missed_block_ranges().unwrap();
-    assert_eq!(missed_block_ranges.len(), 0);
-
-    // Check that inserting an empty block range is rejected and doesn't change things
-    assert!(!db.report_missed_block_range(&BlockRange::new(1, 1)).is_ok());
-
-    let (user_events, _next_start_from_user_event_id) = db.search_user_events(0).unwrap();
-    assert_eq!(user_events.len(), 0);
-
-    let missed_block_ranges = db.get_missed_block_ranges().unwrap();
-    assert_eq!(missed_block_ranges.len(), 0);
-
-    // Check that inserting a block range [1, 3] works and advances num_blocks
-    let ok_block_range = BlockRange::new(1, 3);
-    db.report_missed_block_range(&ok_block_range).unwrap();
-
-    let (user_events, _next_start_from_user_event_id) = db.search_user_events(0).unwrap();
-    assert_eq!(
-        user_events,
-        vec![FogUserEvent::MissingBlocks(ok_block_range.clone())]
-    );
-
-    let missed_block_ranges = db.get_missed_block_ranges().unwrap();
-    assert_eq!(missed_block_ranges, vec![ok_block_range.clone()]);
-
-    // Check that inserting a block range [2, 4] fails because it overlaps
-    assert!(!db.report_missed_block_range(&BlockRange::new(2, 4)).is_ok());
-
-    let (user_events, _next_start_from_user_event_id) = db.search_user_events(0).unwrap();
-    assert_eq!(
-        user_events,
-        vec![FogUserEvent::MissingBlocks(ok_block_range.clone())]
-    );
-
-    let missed_block_ranges = db.get_missed_block_ranges().unwrap();
-    assert_eq!(missed_block_ranges, vec![ok_block_range.clone()]);
-
-    // Check that inserting a block range [0, 4] fails because it overlaps
-    assert!(!db.report_missed_block_range(&BlockRange::new(0, 4)).is_ok());
-
-    let (user_events, _next_start_from_user_event_id) = db.search_user_events(0).unwrap();
-    assert_eq!(
-        user_events,
-        vec![FogUserEvent::MissingBlocks(ok_block_range.clone())]
-    );
+pub fn recovery_db_missed_blocks_reporting(
+    rng: &mut (impl RngCore + CryptoRng),
+    db: &impl RecoveryDb,
+) {
+    let ingress_key = CompressedRistrettoPublic::from(RistrettoPublic::from_random(rng));
+    db.new_ingress_key(&ingress_key, 0).unwrap();
```

Comment:
I would expand this test to see what happens if we have overlapping ingress keys.
Its unlikely, but we might run into it while trying to deploy a new version and somehow losing the entire old and new clusters.

Reply:
done